lib/remarkable_ruby/objects/document.rb

```diff
@@ -0,0 +1,45 @@
+module RemarkableRuby
+  class Document < Object
+    # Download the zip file for a given document in the user's current directory
+    def download
+      params = { doc: @uuid, withBlob: true }
+      response = @connection.get("document-storage/json/2/docs", params)
+
+      dl_link = extract_link(response)
+      streamed = []
+      @connection.get(dl_link) do |req|
+        req.options.on_data = Proc.new { |chunk| streamed << chunk }
+      end
+
+      new_file_name = "#{name}.zip"
+      File.write(new_file_name, streamed.join)
+      new_file_name
+    end
+
+    def delete
+      payload = [{ ID: uuid, Version: version }]
+      response = connection.put("/document-storage/json/2/delete", payload)
+    end
+
+    def highlights
+      highlights = []
+      download unless File.exists?("#{name}.zip")
+      Zip::File.open("#{name}.zip") do |zip_file|
+        zip_file.each do |entry|
```

Comment:
Essentially the same thing, but another way to do this would be to do something like:
`entries_with_highlights = zip_file.select { |entry| entry.name.include?("highlights") }` and then loop through  those entries.

Reply:
Yea that makes more sense. Good idea.